src/pages/[id]/index.js

```diff
@@ -50,11 +47,26 @@ const MemberProfile = ({ imageLink, user, contributions, errorMessage }) => {
     fillActiveTasksArray();
   }, [fillActiveTasksArray]);
 
+  if (errorMessage) {
+    return <NotFound errorMsg={errorMessage} />;
+  }
+
+  if (!isSuperUser) {
+    if (!apiCalledToVerifySuperUser) {
+      makeApiCallToVerifySuperUser();
+    }
+  }
+  if (!isSuperUser && !user.roles?.member) {
+    const errorMsg = `The Member Page for ${
+      id.charAt(0).toUpperCase() + id.slice(1)
+    } is not yet generated.`;
+    return <NotFound errorMsg={errorMsg} />;
+  }
   const { first_name = '', last_name = '' } = user;
   const memberName = `${first_name} ${last_name} | Member Real Dev Squad`;
 
-  const { query } = useRouter(); // this needs to be changed
-  const devUser = !!query.dev;
+  // const { query } = useRouter(); // this needs to be changed
```

Comment:
NIT: If not required we should just remove it

Reply:
sure we can remove this.